apps/op-func/src/infra/storage/access-log-register.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  BlobServiceClient,
+  BlockBlobUploadResponse,
+  ContainerClient,
+} from "@azure/storage-blob";
+import {
+  AuditEvent,
+  auditEventSchema,
+} from "io-fims-common/domain/audit-event";
+import * as assert from "node:assert/strict";
+
+export class AccessLogRegister {
+  #container: ContainerClient;
+
+  constructor(blobServiceClient: BlobServiceClient, containerName: string) {
+    this.#container = blobServiceClient.getContainerClient(containerName);
+  }
+
+  // A helper method used to read a Node.js readable stream into a String
+  async #streamToAuditEvent(readableStream: NodeJS.ReadableStream) {
+    let result = "";
+    for await (const chunk of readableStream) {
+      result += chunk;
+    }
+
+    try {
+      const parsedResult = JSON.parse(result);
+      return auditEventSchema.parse(parsedResult);
+    } catch (e) {
+      throw new Error(`Error during conversion of the downloaded stream`, {
+        cause: e,
+      });
+    }
+  }
+
+  async get(name: string): Promise<AuditEvent> {
+    const blobClient = this.#container.getBlobClient(name);
+
+    // Get blob content from position 0 to the end
+    // get downloaded data by accessing downloadBlockBlobResponse.readableStreamBody
+    const downloadBlockBlobResponse = await blobClient.download();
+    const downloadedStream = downloadBlockBlobResponse.readableStreamBody;
+    assert.ok(downloadedStream, "No Blob with name " + name + " found");
+    try {
+      return await this.#streamToAuditEvent(downloadedStream);
+    } catch (error) {
+      throw new Error(`Error retrieving blob with name ${name}`, {
+        cause: error,
+      });
+    }
+  }
+
+  async upload(content: AuditEvent): Promise<BlockBlobUploadResponse> {
+    const blockBlobClient = this.#container.getBlockBlobClient(
+      content.blobName,
+    );
+    try {
+      const parsedContent = JSON.stringify(content);
+      return blockBlobClient.upload(parsedContent, parsedContent.length);
+    } catch (e) {
```

Comment:
please be consistent on errors names, the previous was named `error`, here it is named `e`